parity/cli/subcommands.rs

```diff
@@ -0,0 +1,242 @@
+use serde_derive::Deserialize;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+pub enum SubCommands {
+    Daemon(Daemon),
+    Wallet {
+        #[structopt(subcommand)]
+        wallet: Wallet,
+    },
+    Account {
+        #[structopt(subcommand)]
+        account: Account,
+    },
+    Import(Import),
+    Export {
+        #[structopt(subcommand)]
+        export: Export,
+    },
+    Signer(Signer),
+    Snapshots(Snapshots),
+    Restore(Restore),
+    Tools(Tools),
+    Db(Db),
+    #[structopt(
+        about = "Print the hashed light clients headers of the given --chain (default: mainnet) in a JSON format. To be used as hardcoded headers in a genesis file."
+    )]
+    ExportHardcodedSync,
+    Dapp(Dapp),
+}
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+#[structopt(about = "Use parity as a daemon")]
+pub struct Daemon {
+    #[structopt(long = "pid-file", name = "PID-FILE", help = "Path to the pid file")]
+    pub pid_file: Option<String>,
+}
+
+#[derive(StructOpt, Deserialize, Debug, Clone)]
+#[structopt(about = "Manage accounts")]
+pub enum Account {
+    #[structopt(
+        about = "Create a new account (and its associated key) for the given --chain [default: mainnet]"
+    )]
+    New,
+    #[structopt(about = "List existing accounts of the given --chain [default: mainnet]")]
+    List,
+    #[structopt(
+        about = "Import accounts from JSON UTC keystore files to the specified --chain [default: mainnet]"
+    )]
+    // FIXME: The original parser implementation had this as `Option<Vec<String>>` but this is not
+    // supported by structopt yet, referring to issue
+    // [#364](https://github.com/TeXitoi/structopt/issues/364)
```

Comment:
Hmm, that's not quite how I read #364 – it seems to be quirky but not unsupported. Can you elaborate?

Reply:
I believe the empty list was treated equivalently to `None` in the previous version, and it's the same in this version. All the previous tests are passing, so unless this was a case not tested, we are good